Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -1734,6 +1954,126 @@ private void UpdateRigHandles()
                 rigRoot.parent = transform;
             }
         }
+
+        private HandleProximityState ScaleHandleByProximity(Transform handle, Renderer renderer, HandleType type, HandleProximityState state, List<Vector3> points, float defaultScale)
+        {
+            HandleProximityState newState = state;
+            float closestDistanceSqr = float.MaxValue;
+            foreach (Vector3 point in points)
+            {
+                closestDistanceSqr = Mathf.Min(closestDistanceSqr,(handle.position - point).sqrMagnitude);
+            }
+
+            if (closestDistanceSqr < handleCloseProximity)
+            {
+                newState = HandleProximityState.CloseProximity;
+            }
+            else if (closestDistanceSqr < handleMediumProximity)
+            {
+                newState = HandleProximityState.MediumProximity;
+            }
+            else//far
+            {
+                newState = HandleProximityState.FullsizeNoProximity;
+            }
+
+            ScaleHandle(handle, newState, defaultScale, type);
+
+            if (renderer != null && state != newState)
+            {
+                renderer.material = newState == HandleProximityState.CloseProximity ? handleGrabbedMaterial : handleMaterial;
+            }
+            return newState;
+        }
+
+        private void ScaleHandle(Transform handle, HandleProximityState targetState, float defaultScale, HandleType type)
+        {
+            float targetScale = targetState == HandleProximityState.CloseProximity ? closeScale : (targetState == HandleProximityState.MediumProximity ? mediumScale : farScale);
+            float weight = (targetState == HandleProximityState.CloseProximity ? closeGrowRate : (targetState == HandleProximityState.MediumProximity ? mediumGrowRate : farGrowRate));
+            float newLocalScale = (handle.transform.localScale.x * (1.0f - weight)) + (scaleHandleSize * targetScale * weight) * defaultScale;
+            handle.transform.localScale = new Vector3(newLocalScale, newLocalScale, newLocalScale);
+
+            Collider collider = handle.gameObject.GetComponent<Collider>();
+
+            float colliderScalar = type == HandleType.Rotation ? rotationHandleColliderDiameter : cornerHandleColliderScale;
+            if (collider && collider is BoxCollider)
+            {
+                newLocalScale = Mathf.Abs(newLocalScale);
+                newLocalScale = newLocalScale == 0.0f ? 0.01f : newLocalScale;
+                float colliderScale = colliderScalar * 0.02f;
```

Comment:
It would be nice to avoid magic numbers like this, and in the line above. Can you explain why you have this multiplier here?

Reply:
its just to keep the numbers from getting really small for the user. These could be removed.